controllers/securitypolicy_controller.go

```diff
@@ -140,33 +140,48 @@ func (r *SecurityPolicyReconciler) spec(
 // updateAccess updates given AccessDefinition's APIID and APIName fields based on ApiDefinition CR that is referred
 // in the AccessDefinition. So that, it includes k8s details of the referred ApiDefinitions.
 func (r *SecurityPolicyReconciler) updateAccess(ctx context.Context, ad *tykv1.AccessDefinition) error {
+	var apiID, apiName string
 	api := &tykv1.ApiDefinition{}
 	if err := r.Get(ctx, types.NamespacedName{Name: ad.Name, Namespace: ad.Namespace}, api); err != nil {
 		if errors.IsNotFound(err) {
-			r.Log.Info("ApiDefinition resource not found. Unable to attach to SecurityPolicy. ReQueue",
+			r.Log.Info("ApiDefinition resource not found. Unable to attach to SecurityPolicy. looking for OasApiDefinition",
 				"Name", ad.Name,
 				"Namespace", ad.Namespace,
 			)
-
-			return err
 		}
+	}
 
-		r.Log.Error(err, "Failed to get APIDefinition to attach to SecurityPolicy")
+	apiID = api.Status.ApiID
+	apiName = api.Spec.Name
 
-		return err
+	if apiID == "" {
+		r.Log.Info("Failed to get normal APIDefinition to attach to SecurityPolicy, looking for OasApiDefiniiton")
+
+		oasApi := &tykv1.TykOASApiDefinition{}
+
+		if errOas := r.Get(ctx, types.NamespacedName{Name: ad.Name, Namespace: ad.Namespace}, oasApi); errOas != nil {
+			if errors.IsNotFound(errOas) {
+				r.Log.Info("OasApiDefinition resource also not found. Unable to attach to SecurityPolicy. ReQueue",
+					"Name", ad.Name,
+					"Namespace", ad.Namespace,
+				)
+			}
+		}
+
+		apiID = oasApi.Status.ApiID
+		apiName = oasApi.Name
 	}
 
-	if api.Status.ApiID == "" {
+	if apiID == "" {
 		r.Log.Error(
 			opclient.ErrNotFound,
 			"Failed to find ApiDefinition on Tyk", "ApiDefinition", client.ObjectKeyFromObject(api),
 		)
-
 		return opclient.ErrNotFound
```

Comment:
Can't we return early if we couldn't find OAS ApiDef [above](https://github.com/TykTechnologies/tyk-operator/pull/579/files#diff-94d85e85cce9ea50d75136697ae3954462fe96beb26ee1030a7c385cd75ed572R163-R167)?

Reply:
**Yes we can, fixed.**